request-example.go

```diff
@@ -151,16 +171,8 @@ func (fs *root) Filecmd(r *Request) error {
 
 	switch r.Method {
 	case "Setstat":
-		file, err := fs.openfile(r.Filepath, sshFxfWrite)
-		if err != nil {
-			return err
-		}
-
-		if r.AttrFlags().Size {
-			return file.Truncate(int64(r.Attributes().Size))
-		}
-
-		return nil
+		flags := r.AttrFlags()
+		return fs.SetStat(r, &flags, r.Attributes())
```

Comment:
By making `SetStat()` take a `FileAttrFlags` instead of `*FileAttrFlags` then there there is no need to assign the value to a temporary value, and thus we avoid one of the 2 hardest things in programming:

1. Naming things
2. Cache coherency
3. Off-by-one errors